src/main/java/capstone/mfslbackend/service/TransactionService.java

```diff
@@ -55,9 +62,56 @@ public Transaction createTransaction(Long fantasyTeamId, Long incomingPlayerId,
+                }
+            } else if (position.equals("Defender")) {
+                if (proposingFantasyTeam.getPlayers().stream().filter(player -> player.getPosition().equals("Defender")).count() == MAX_DEF && !outgoingPlayer.getPosition().equals("Defender")) {
+                        transaction.setStatus(TransactionStatus.REJECTED);
+                        throw new Error400("Fantasy team with id " + fantasyTeamId + " already has the maximum number of defenders");
+                }  else if (proposingFantasyTeam.getPlayers().stream().filter(player -> player.getPosition().equals("Defender")).count() <= MIN_DEF && outgoingPlayer.getPosition().equals("Defender")) {
+                transaction.setStatus(TransactionStatus.REJECTED);
+                throw new Error400("Fantasy team with id " + fantasyTeamId + " needs at least 3 defenders");
+            }
+            } else if (position.equals("Midfielder")) {
+                if (proposingFantasyTeam.getPlayers().stream().filter(player -> player.getPosition().equals("Midfielder")).count() == MAX_MID && !outgoingPlayer.getPosition().equals("Midfielder")) {
+                        transaction.setStatus(TransactionStatus.REJECTED);
+                        throw new Error400("Fantasy team with id " + fantasyTeamId + " already has the maximum number of midfielders");
+                } else if (proposingFantasyTeam.getPlayers().stream().filter(player -> player.getPosition().equals("Midfielder")).count() <= MIN_MID && outgoingPlayer.getPosition().equals("Midfielder")) {
+                transaction.setStatus(TransactionStatus.REJECTED);
+                throw new Error400("Fantasy team with id " + fantasyTeamId + " needs at least 3 midfielders");
+            }
+            } else if (position.equals("Attacker")) {
+                if (proposingFantasyTeam.getPlayers().stream().filter(player -> player.getPosition().equals("Attacker")).count() == MAX_ATT && !outgoingPlayer.getPosition().equals("Attacker")) {
+                    transaction.setStatus(TransactionStatus.REJECTED);
+                    throw new Error400("Fantasy team with id " + fantasyTeamId + " already has the maximum number of attackers");
+                } else if (proposingFantasyTeam.getPlayers().stream().filter(player -> player.getPosition().equals("Attacker")).count() <= MIN_ATT && outgoingPlayer.getPosition().equals("Attacker")) {
+                    transaction.setStatus(TransactionStatus.REJECTED);
+                    throw new Error400("Fantasy team with id " + fantasyTeamId + " needs at least 1 attacker");
+                }
+            }
+
+            //if it satisfies minimum and max requirements, the transaction is accepted
+            String replacement = proposingFantasyTeam.getPlayerIdsInOrder().replace(outgoingPlayerId.toString(), incomingPlayerId.toString());
+            proposingFantasyTeam.setPlayerIdsInOrder(replacement);
```

Comment:
here you setPlayerIdsInOrder(), but you also have to setPlayers() with the new player in the list